transports/tcp/src/lib.rs

```diff
@@ -1240,4 +1258,43 @@ mod tests {
 
         test("/ip4/127.0.0.1/tcp/12345/tcp/12345".parse().unwrap());
     }
+
+    #[cfg(any(feature = "async-io", feature = "tcp"))]
+    #[test]
+    fn test_address_translation() {
+        #[cfg(feature = "async-io")]
+        let transport = TcpTransport::new(GenTcpConfig::new());
+        #[cfg(all(feature = "tokio", not(feature = "async-io")))]
+        let transport = TokioTcpTransport::new(GenTcpConfig::new());
+
+        let port = 42;
+        let tcp_listen_addr = Multiaddr::empty()
+            .with(Protocol::Ip4(Ipv4Addr::new(127, 0, 0, 1)))
+            .with(Protocol::Tcp(port));
```

Comment:
I always find these easier to read if we parse them from a string.

Reply:
I prefer using types where possible, avoiding any unwraps that would be needed when parsing from string :)